src/ProjectSystem/Impl/IO/MsBuildFileSystemWatcher.cs

```diff
@@ -215,6 +215,9 @@ public sealed partial class MsBuildFileSystemWatcher : IDisposable {
 
             relativePath = PathHelper.MakeRelative(rootDirectory, fullPath);
             try {
+                if (!fileSystem.FileExists(fullPath)) {
```

Comment:
Ah, I think we are talking 3 different things since I found 3 different issues :-)

1. IsFileAllowed throwing - no popup, but I've seen file added but not showing up in Solution Explorer.
2. CPS 'not supported' exception when 'Link' metadata mismatch.
3. The popup in the beginning of the bug - yes, that is about ToShortRelativePath and I am going to look into it.